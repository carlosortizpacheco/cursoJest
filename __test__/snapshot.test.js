import { getCharacter } from "../snapshot"
import rick from "../rick.json"

describe('Trabajando con instantaneas', () => {
    test('Snapshot', () => {
        expect(getCharacter(rick)).toMatchSnapshot()
    });
    test("siempre fallara el snapshot", () => {
        const user = {
            createAt: new Date(),
            id: Math.floor(Math.random() * 20),
        }
        expect(user).toMatchSnapshot();
    });
    test("Permitimos una excepcion", () => {
        const user = {
            id: Math.floor(Math.random() * 20),
            name: 'Carlos Ortiz'
        }
        expect(user).toMatchSnapshot({
            id: expect.any(Number)
        });
    });
});



