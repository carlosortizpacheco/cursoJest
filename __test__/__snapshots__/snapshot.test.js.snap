// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trabajando con instantaneas Permitimos una excepcion 1`] = `
Object {
  "id": Any<Number>,
  "name": "Carlos Ortiz",
}
`;

exports[`Trabajando con instantaneas Snapshot 1`] = `
Object {
  "gender": "male",
  "id": 1,
  "name": "Rick Sanches",
  "species": "human",
  "status": "Alive",
}
`;

exports[`Trabajando con instantaneas siempre fallara el snapshot 1`] = `
Object {
  "createAt": 2020-08-20T20:22:02.029Z,
  "id": 15,
}
`;
